src/beta/resources/referral.js

```diff
@@ -0,0 +1,153 @@
+/* eslint-disable no-dupe-class-members,no-unused-vars,class-methods-use-this */
+import superagent from 'superagent';
+
+import T from 'proptypes';
+import base from '../../resources/base';
+import { propTypes as userPropTypes } from '../../resources/user';
+import RequestError from '../../errors/request';
+import API from '../../easypost';
+
+export const propTypes = Object.assign({}, userPropTypes, {
+  name: T.string.isRequired,
+  email: T.string.isRequired,
+  phone: T.string.isRequired,
+});
+
+const stripeCreditCardUrl = 'https://api.stripe.com/v1/tokens';
+
+/**
+ * Get an instance of the API client using the referral user's API key.
+ * @param {API} api - The API client to copy.
+ * @param {string} referralApiKey - The referral user's API key.
+ * @returns {API} - An instance of the API client.
+ */
+function getReferralApi(api, referralApiKey) {
+  return API.copyApi(api, {
+    apiKey: referralApiKey,
+  });
+}
+
+/**
+ * Get EasyPost's Stripe API key used to create credit cards on Stripe's servers.
+ * @param {API} api - The API client to use.
+ * @returns {string} - The Stripe API key.
+ */
+async function getEasyPostStripeKey(api) {
+  const response = await api.get('partners/stripe_public_key');
+
+  return response.body.public_key;
+}
+
+/**
+ * Send the credit card details to Stripe to get a Stripe credit card token.
+ * @param {string} stripeKey - The Stripe API key.
+ * @param {string} number - Credit card number.
+ * @param {string} expirationMonth - Credit card expiration month.
+ * @param {string} expirationYear - Credit card expiration year.
+ * @param {string} cvc - Credit card CVC.
+ * @returns {Promise<string>} - Stripe credit card ID.
+ */
+async function sendCardDetailsToStripe(stripeKey, number, expirationMonth, expirationYear, cvc) {
+  // need to form-encode per Stripe's API
+  const request = superagent.post(stripeCreditCardUrl).set({
+    Authorization: `Bearer ${stripeKey}`,
+    'Content-Type': 'application/x-www-form-urlencoded',
+  });
+  request.send({
+    card: {
+      number,
+      exp_month: expirationMonth,
+      exp_year: expirationYear,
+      cvc,
+    },
+  });
+
+  try {
+    const response = await request;
+
+    return response.body.id;
+  } catch (error) {
+    if (error.response && error.response.body) {
+      throw new RequestError(error.response.body, stripeCreditCardUrl);
```

Comment:
Nit: did we want to match the Python lib's approach where we return our own message since whatever stripe returns us is out of our control?

Reply:
Sure